trace.go

```diff
@@ -234,6 +234,21 @@ func NewAggregate(errs ...error) error {
 	return wrap(aggregate(errs), 2)
 }
 
+// NewAggregateFromChannel creates a new aggregate instance from the provided
+// errors channel.
+//
+// It is the caller's responsibility to close the channel, otherwise this
+// function might block!
+func NewAggregateFromChannel(errCh chan error) error {
```

Comment:
what worries me in this implementation there's no ability to cancel it. I would give it a `net/context` as optional parameter so you could cancel this operation if needed